oak_runtime/src/node/http/mod.rs

```diff
@@ -465,56 +528,139 @@ fn get_oak_label(req: &HttpRequest) -> Result<Label, OakStatus> {
+                .map(|m| m.values.as_slice())
+        }),
+        req.headers.as_ref().and_then(|map| {
+            map.headers
+                .get(oak_abi::OAK_SIGNED_CHALLENGE_PROTOBUF_KEY)
+                .map(|m| m.values.as_slice())
+        }),
+    );
+
+    match headers {
+        (Some([json_signature]), None) => verify_json_challenge(json_signature),
+        (None, Some([protobuf_signature])) => verify_protobuf_challenge(protobuf_signature),
+        _ => Ok(vec![]),
     }
 }
 
-fn parse_json_label(label_str: &[u8]) -> Result<Label, OakStatus> {
+fn parse_json_label(label_str: &[u8]) -> Result<Label, HttpError> {
     let label_str = String::from_utf8(label_str.to_vec()).map_err(|err| {
-        warn!(
+        HttpError::RequestLabelError(format!(
             "The label must be a valid UTF-8 JSON-formatted string: {}",
             err
-        );
-        OakStatus::ErrInvalidArgs
+        ))
     })?;
-    serde_json::from_str(&label_str).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
-    })
+    serde_json::from_str(&label_str)
+        .map_err(|err| HttpError::RequestLabelError(format!("Could not parse HTTP label: {}", err)))
 }
 
-fn parse_protobuf_label(base64_protobuf_label: &[u8]) -> Result<Label, OakStatus> {
+fn parse_protobuf_label(base64_protobuf_label: &[u8]) -> Result<Label, HttpError> {
     let protobuf_label = base64::decode(base64_protobuf_label).map_err(|err| {
-        warn!("Could not decode Base64 HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
+        HttpError::RequestLabelError(format!("Could not decode Base64 HTTP label: {}", err))
+    })?;
+    Label::decode(&protobuf_label[..])
+        .map_err(|err| HttpError::RequestLabelError(format!("Could not parse HTTP label: {}", err)))
+}
+
+fn verify_json_challenge(signature: &[u8]) -> Result<Vec<u8>, HttpError> {
+    let signature = parse_json_signed_challenge(signature.to_vec()).map_err(|err| {
+        HttpError::IdentityVerificationError(format!(
+            "Could not parse json formatted signed challenge: {:?}",
+            err
+        ))
     })?;
-    Label::decode(&protobuf_label[..]).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
+    verify_signed_challenge(signature)
+}
+
+fn parse_json_signed_challenge(
+    bytes: Vec<u8>,
+) -> Result<oak_abi::proto::oak::identity::SignedChallenge, HttpError> {
+    let signature_str = String::from_utf8(bytes).map_err(|err| {
+        HttpError::IdentityVerificationError(format!("Could not parse signed challenge: {:?}", err))
+    })?;
+    serde_json::from_str(&signature_str).map_err(|err| {
+        HttpError::IdentityVerificationError(format!("Malformed signed challenge: {:?}", err))
     })
 }
 
-struct HttpResponseIterator {
+fn verify_protobuf_challenge(base64_signature: &[u8]) -> Result<Vec<u8>, HttpError> {
```

Comment:
Could you document the return types of these methods? They all return bytes, so they are not very self-explanatory.

Reply:
Added some documentation.